server/src/main/java/io/druid/server/coordinator/DiskNormalizedCostBalancerStrategy.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.coordinator;
+
+import com.google.common.base.Predicates;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.metamx.emitter.EmittingLogger;
+import io.druid.java.util.common.Pair;
+import io.druid.timeline.DataSegment;
+import org.apache.commons.math3.util.FastMath;
+import org.joda.time.Interval;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+
+public class DiskNormalizedCostBalancerStrategy extends CostBalancerStrategy
+{
+  public DiskNormalizedCostBalancerStrategy(ListeningExecutorService exec)
+  {
+    super(exec);
+  }
+
+  @Override
+  protected double computeCost(
+      final DataSegment proposalSegment, final ServerHolder server, final boolean includeCurrentServer
+  )
+  {
+    double cost = super.computeCost(proposalSegment, server, includeCurrentServer);
+
+    if(cost == Double.POSITIVE_INFINITY){
+      return cost;
+    }
+
+    int nSegments = 1;
+    if(server.getServer().getSegments().size() > 0)
+    {
+      nSegments = server.getServer().getSegments().size();
+    }
+
+    double normalizedCost = cost/nSegments;
+    double usageRatio = (double)server.getServer().getCurrSize()/(double)server.getServer().getMaxSize();
+
+    return normalizedCost*usageRatio;
```

Comment:
I see, technically cost is not proportional to the number of segments, but if your segments all cover the same interval length, using that approximation (essentially multiplying your cost by the average segment size) might work.

I've thought about it a little bit more and I was thinking a more consistent approach could be to add a "weight" factor to the existing cost balancer, where we assigns each segment a weight proportional to the size of the segment. Essentially if we just added two parameters `double weightA`, `double weight B` to [`computeJointSegmentsCost(final DataSegment segmentA, final DataSegment segmentB)`](https://github.com/druid-io/druid/blob/master/server/src/main/java/io/druid/server/coordinator/CostBalancerStrategy.java#L64) and expand the multiplier to be `multiplier = multiplier * weightA * weightB`, then you should only have to divide the resulting cost by the server maxSize to achieve your objective.

I don't want to hold up this PR any longer, but how hard would it be to try this out and see if you get similar results in your setup? As long as we can commit to trying it out, I think we can merge what we have right now, so we can get the refactoring in.